src/PrestaShopBundle/Form/Admin/Sell/Product/Category/CategoryFilterType.php

```diff
@@ -0,0 +1,139 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Admin\Sell\Product\Category;
+
+use PrestaShop\PrestaShop\Adapter\Category\CategoryDataProvider;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * This form type is used in the product list, it displays a tree to select one category used to filter the list.
+ */
+class CategoryFilterType extends AbstractType
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var CategoryDataProvider
+     */
+    private $categoryProvider;
+
+    /**
+     * @var int
+     */
+    private $languageId;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param CategoryDataProvider $categoryDataProvider
+     * @param int $languageId
+     */
+    public function __construct(
+        TranslatorInterface $translator,
+        CategoryDataProvider $categoryDataProvider,
+        int $languageId
+    ) {
+        $this->translator = $translator;
+        $this->categoryProvider = $categoryDataProvider;
+        $this->languageId = $languageId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        parent::buildView($view, $form, $options);
+        $view->vars['nested_tree'] = $options['nested_tree'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+        $nestedTree = $this->categoryProvider->getNestedCategories(null, $this->languageId, false);
+        $flattenedTree = $this->flattenTree($nestedTree);
+
+        $resolver->setDefaults([
+            'label' => $this->translator->trans('Categories', [], 'Admin.Catalog.Feature'),
+            'choices' => $flattenedTree,
+            'nested_tree' => $nestedTree,
+            'multiple' => false,
+            'expanded' => false,
+            'required' => false,
+            'choice_label' => 'name',
+            'choice_value' => 'category_id',
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParent(): string
```

Comment:
I was thinking no method have return hint type .... but this one have a ": string" . Do we have to add return type to others or remove this one ? I don't know.

Reply:
I would suggest leaving inheritdoc and simply following interfaces. If interface doesn't have the type, so don't add it here either.
OR
if we decide following docblock typehints of interfaces already, then actual types should match, meaning this method should return `?string` no?
Probably worth discussing and deciding. Up until now I think we had kind of agreement to leave "@inheritdoc" on such classes (that implements older interfaces which has no strict types)

Reply:
Yeah we usually respect the prototype from the framework I didn't pay attention, I either copied this or let the IDE create it for me I don't remember but I'll check and remove the type if necessary

Reply:
I have no opinion. Sure I prefer an high level of typehinting (I wait for the 7.4 as minimal version to force the typehinting of properties) but we are in a middleware right now. No matter to leave it like it is. I was just for open the debate.

Reply:
The Symfony methods have no strong typing, so we don't add one here either